src/main.js

```diff
@@ -30,72 +30,150 @@ app.whenReady().then(() => {
 });
 
 app.on("window-all-closed", () => {
-    if (process.platform !== "darwin")
-        app.quit();
+    if (process.platform !== "darwin") app.quit();
 });
 
 function createWindow() {
     const window = new BrowserWindow({
         webPreferences: {
             preload: path.join(__dirname, "preload.js"),
             // Allow the preload script to read local files
-            sandbox: false
+            sandbox: false,
         },
-        show: false
+        show: false,
     });
     window.maximize();
     window.loadFile(path.join(__dirname, "renderer/index.html"));
     window.show();
 }
 
 function createApplicationMenu() {
+    const isMac = process.platform === "darwin";
+
     const menu = Menu.buildFromTemplate([
+        // { role: 'appMenu' }
+        ...(isMac
+            ? [
+                {
+                    label: app.name,
+                    submenu: [
+                        { role: "about" },
+                        { type: "separator" },
+                        { role: "services" },
+                        { type: "separator" },
+                        { role: "hide" },
+                        { role: "hideOthers" },
+                        { role: "unhide" },
+                        { type: "separator" },
+                        { role: "quit" },
+                    ],
+                },
+            ]
+            : []),
+        // { role: 'fileMenu' }
         {
             label: "&File",
             submenu: [
                 {
-                    label: "Open",
+                    label: isMac ? "Open..." : "Open",
                     accelerator: "CmdOrCtrl+O",
                     click: (_, browserWindow) => {
                         browserWindow.webContents.send("show-open-files-view");
-                    }
+                    },
                 },
-                { role: "quit" }
-            ]
+                isMac ? { role: "close" } : { role: "quit" },
+            ],
         },
+        // { role: 'editMenu' }
         {
-            label: "&View",
+            label: "&Edit",
+            submenu: [
+                { role: "undo" },
+                { role: "redo" },
+                { type: "separator" },
+                { role: "cut" },
+                { role: "copy" },
+                { role: "paste" },
+                ...(isMac
+                    ? [
+                        { role: "pasteAndMatchStyle" },
+                        { role: "delete" },
+                        { role: "selectAll" },
+                        { type: "separator" },
+                        {
+                            label: "Speech",
+                            submenu: [
+                                { role: "startSpeaking" },
```

Comment:
Is this for navigating the UI with speech? Does it actually work?

Reply:
Yes it does :)